media-proxy/include/mesh/st2110rx.h

```diff
@@ -0,0 +1,174 @@
+#ifndef ST2110RX_H
+#define ST2110RX_H
+
+#include "st2110.h"
+
+namespace mesh::connection {
+
+/**
+ * ST2110Rx
+ *
+ * Base abstract class of ST2110Rx. ST2110_20Rx/ST2110_22Rx/ST2110_30Rx
+ * inherit this class.
+ */
+template <typename FRAME, typename HANDLE, typename OPS> class ST2110Rx : public ST2110 {
+  public:
+    ST2110Rx() : mtl_session(nullptr), ops{0}, transfer_size(0) { _kind = Kind::receiver; }
+
+    ~ST2110Rx() {
+        shutdown(_ctx);
+        if (ops.name)
+            free((void *)ops.name);
+    }
+
+  protected:
+    HANDLE mtl_session;
+    OPS ops;
+    size_t transfer_size;
+    std::jthread frame_thread_handle;
+    context::Context _ctx = context::WithCancel(context::Background());
+
+    virtual FRAME *get_frame(HANDLE) = 0;
+    virtual int put_frame(HANDLE, FRAME *) = 0;
+    virtual HANDLE create_session(mtl_handle, OPS *) = 0;
+    virtual int close_session(HANDLE) = 0;
+
+    int configure_common(context::Context& ctx, const std::string& dev_port,
+                         const MeshConfig_ST2110& cfg_st2110) {
+        int session_id = 0;
+        mtl_device = get_mtl_device(dev_port, MTL_LOG_LEVEL_CRIT, cfg_st2110.local_ip_addr, session_id);
+        if (mtl_device == nullptr) {
+            log::error("Failed to get MTL device");
+            return -1;
+        }
+
+        inet_pton(AF_INET, cfg_st2110.remote_ip_addr, ops.port.ip_addr[MTL_PORT_P]);
+        inet_pton(AF_INET, cfg_st2110.local_ip_addr, ops.port.mcast_sip_addr[MTL_PORT_P]);
+        ops.port.udp_port[MTL_PORT_P] = cfg_st2110.local_port;
+
+        strlcpy(ops.port.port[MTL_PORT_P], dev_port.c_str(), MTL_PORT_MAX_LEN);
+        ops.port.num_port = 1;
+
+        char session_name[NAME_MAX] = "";
+        snprintf(session_name, NAME_MAX, "mcm_mtl_rx_%d", session_id);
+        if (ops.name)
+            free((void *)ops.name);
+        ops.name = strdup(session_name);
+        ops.framebuff_cnt = 4;
+
+        ops.priv = this; // app handle register to lib
+        ops.notify_frame_available = frame_available_cb;
+
+        log::info("ST2110Rx: configure")("port", ops.port.port[MTL_PORT_P])(
+            "ip_addr", std::to_string(ops.port.ip_addr[MTL_PORT_P][0]) + " " +
```

Comment:
Please rework logging everywhere to start the line with the opening bracket `(` at the begining of each line. This should be applied to the logging parameters only.
```suggestion
        log::info("ST2110Rx: configure")("port", ops.port.port[MTL_PORT_P])
            ("ip_addr", std::to_string(ops.port.ip_addr[MTL_PORT_P][0]) + " " +
```

Reply:
Fixed in https://github.com/OpenVisualCloud/Media-Communications-Mesh/pull/261/commits/92513fe839a25353cd6ab4ced946b0ab5ed474c9